app/controllers/consolidations/ShutMucrSummaryController.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.consolidations
+
+import controllers.actions.{AuthAction, JourneyAction}
+import controllers.exception.IncompleteApplication
+import controllers.storage.CacheIdGenerator.movementCacheId
+import controllers.storage.FlashKeys
+import forms.ShutMucr
+import javax.inject.Inject
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.{CustomsCacheService, SubmissionService}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.shut_mucr_summary
+
+import scala.concurrent.ExecutionContext
+
+class ShutMucrSummaryController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  mcc: MessagesControllerComponents,
+  cacheService: CustomsCacheService,
+  submissionService: SubmissionService,
+  shutMucrSummaryPage: shut_mucr_summary
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def displayPage(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    cacheService.fetchAndGetEntry[ShutMucr](movementCacheId(), ShutMucr.formId).map {
+      case Some(data) => Ok(shutMucrSummaryPage(data))
+      case None       => throw IncompleteApplication
+    }
+  }
+
+  def submit(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    cacheService.fetchAndGetEntry[ShutMucr](movementCacheId(), ShutMucr.formId).flatMap {
+      case Some(shutMucr) =>
+        submissionService.submitShutMucrRequest(shutMucr, request.eori).flatMap { _ =>
+          cacheService.remove(movementCacheId()).map { _ =>
```

Comment:
Is removing from keystore is mandatory to make redirection ? I have i mind something like
```
submissionService.submitShutMucrRequest(shutMucr, request.eori).flatMap { _ =>
  Redirect(routes.ShutMucrConfirmationController.displayPage())
              .flashing(FlashKeys.MUCR -> shutMucr.mucr)
}.andThen {
  case Success(_) => cacheService.remove(movementCacheId())
}
```

Reply:
I can change it, I took it from old implementation probably :)

Reply:
Fixed